PR 13836 changes (base on PR 13773)

sunxiaoxia2022 opened the pull request:
### Details:
 - *https://github.com/openvinotoolkit/openvino/pull/13773*
 - https://github.com/openvinotoolkit/openvino/pull/13836

### Tickets:
 - *ticket-id*

rkazants committed:
[TF FE] Refactor translators for Reverse, ReverseV2 and test it (#13602)

* [TF FE] Refactor translators for Reverse, ReverseV2 and test it

Make these operations reshapeable. Add layer tests for them to the pre-commit

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

* Apply code-review feedback: simplify checks in Reverse

* Apply the rest of code-review feedback: simplify code for Reverse

* Remove redundant check for axes

* Apply code-review feedback: support dynamic rank

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Commit:
[GPU] Fix IRDFT output rank. (#13630)

Commit:
program_node refactoring (#13576)

jade-cho committed:
[GPU] Fix a bug that occurred when the optimized layer is executed. (#13568)

jade-cho committed:
[GPU][DG2] Select cldnn reduce kernel when dtype of input/output is FP32. (#13747)

rkazants committed:
[TF FE] Add translator for IsFinite and test it (#13728)

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

rkazants committed:
[TF FE] Refactor translators for Select and SelectV2 and test it (#13699)

* [TF FE] Refactor translators for Select and SelectV2 and test it

It fixes a case when the condition of a smaller rank than operands in Select
Separately added tests for Select and SelectV2
Do not mix-up Select with Where, so tests for Where are moved to test_tf_Where.py

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

* Revert extra changes

* Apply code-review feedback: support undefined rank

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>
Co-authored-by: Andrei Kochin <andrei.kochin@intel.com>

tomdol committed:
Unique-10 operator (#13532)

ilya-lavrenov committed:
Improvements in rpm / debian build (#13750)

* Improvements in rpm / debian build

* Fixed several debian warnings

* Supported old gflags from CentOS 7

* Reverted back OpenCV version

* Fixed clang-format

Commit:
[GPU] Baseline for enabling GPUs from other vendors (#12577)

e-ddykim committed:
[GPU] copying required information at inference time to primitive_inst and kernel_impl_params (#13733)

* usage of program and program_node are changed to use kernerl_impl_params

* pointer to network is replaced with reference

* moved primitive specific parameters out of kernel_impl_params

* use impl_params instead of program_node in onednn_impl

* relocate params from impl_params to primitive_inst

* added conditional directives around fused_desc_onednn

* removal of unnecessary changes

* node_output is separated from impl_param->output

andrei-kochin committed:
[MO TESTS] replace Caffe2 backend with ONNX RT (#13759)

* [MO Tests] Replace Caffe2 backend with ONNX RT

* fix broken PReLU test - broadcastable shape

* xfail GPU test

ilya-lavrenov committed:
Removed duplicated files from OpenVINO wheel (#13725)

eaidova committed:
[OMZ]: relax torch dependencies (#13762)

Co-authored-by: Ilya Lavrenov <ilya.lavrenov@intel.com>

ahnyoung-paul committed:
[GPU] Allow crop optimization for static shape(#13710) (#13710)

ozhydkov-lohika committed:
[GPU] Gather tree blocked layouts support (#12803)

* added gather tree blocked layout support

* gather tree blocked layout support review cleanup

* gather tree blocked layout support review cleanup

* gather tree blocked layout support review cleanup

* gather tree blocked layout support review cleanup

* gather tree blocked layout support review cleanup

* build fixed

ozhydkov-lohika committed:
[GPU] Tile blocked layouts support (#12480)

* add tile blocked layouts support

* build fixed

slyalin committed:
Update add_openvino_ops.md (#13768)

More clear direction for Python users who face the need to implement a new op.

sshlyapn committed:
[GPU] Default context for each device (#13658)

* [GPU] Add default context for each device

* [GPU] Replace operator== overloading with independent function

luo-cheng2021 committed:
disable gen control flow testcases (#13732)

sgolebiewski-intel committed:
DOCS: Replace image in Protecting Model Guide - for master (#13631)

* Replacing image with a scalable svg file

Replacing image in Protecting Model Guide -changing the png image to scalable svg format.
Removing image project files.

sgolebiewski-intel committed:
Fixing indentation in Build Plugin Using CMake (#13477)

Minor fixes that correct indentation of code snippets and note admonitions.

Co-authored-by: Maciej Smyk <maciejx.smyk@intel.com>

msmykx-intel committed:
DOCS: Model Optimizer Usage fix port to master (#13454)

* ref link fix

sgolebiewski-intel committed:
Model-Formats-fix (#13386)

Standardized Additional Resources section for Supported Model Formats along with fixing some ref links in articles.

sgolebiewski-intel committed:
Fixing list in Deployment Manager Tool - for 22.2 (#13379)

A minor fix that corrects the numbered list of steps in Deploying Package on Target Systems

msmykx-intel committed:
DOCS: Port to master - fix for archive installation docs (#13298)

* Fix

* Update installing-openvino-from-archive-linux.md

* fix

* Python 3.10 fix

sgolebiewski-intel committed:
Fixing snippet in Optimization for Throughput - for master (#13378)

A minor fix that corrects the code snippets.

Porting:
https://github.com/openvinotoolkit/openvino/pull/13341

sgolebiewski-intel committed:
DOCS:  API 2.0 Inference Pipeline Fix - for master (#13384)

* Update common_inference_pipeline.md

mateusztabaka committed:
[GPU] Insert transpose for gemm input if its trans attr is set (#8460)

If gemm input dimensions are not multiple of 16 and any of
transpose_a/transpose_b attribute is set - cldnn picks
'gemm_ref' kernel in favor of faster 'gemm_tiled_opt'.
By emplacing explicit permute operation on the gemm input that it requires,
we make cldnn to pick 'gemm_tiled_opt', which in result improves
performance.
For some input shapes, transpose(s) + gemm_tiled_opt can be slower than
just gemm_ref. Based on benchmarks - the cutoff point was set for inputs shapes > (64, 64).

Ticket: 67271

sgolebiewski-intel committed:
Fix pot best practices for master (#13304)

A minor fix that corrects an ordered list in "Tuning Hyperparameters of the Default Quantization" section.

sgolebiewski-intel committed:
Fixing indentation in Local Distribution - for master (#13376)

A minor fix that corrects the indentation of note admonition in the unordered list of Libraries for Execution Modes.

Porting:
https://github.com/openvinotoolkit/openvino/pull/13338

kblaszczak-intel committed:
adjust footer content to meet legal requirement (#13720)

kbruniec committed:
[SAMPLES] [GNA] Remove port assertion in speech_sample when importing (#13687)

* [SAMPLES] [GNA] Remove port assertion in speech_sample when importing the GNA fromat model

* Fix code style

Commit:
[GPU] Matrix NMS (#13137)

a-sidorova committed:
[Snippets] Fixed input and output original precisions in Subgraph (#13451)

EdjeElectronics committed:
Docs: Update "Install OpenVINO Runtime on Linux from Archive File" page (#13345)

* docs: begin updates to install instructions

* docs: change instructions for downloading and unpacking archive

* docs: finish changes to Step 2 and Step 3

* docs: change whats next section

* docs: minor updates and fix typos

* docs: change link to GitHub archive page

* docs: typo fix

* docs: minor change

* docs: replace wget command with curl

* docs: clarify archive install is for C++ users

* docs: add link to PyPI page

* Update docs/install_guides/installing-openvino-from-archive-linux.md

Co-authored-by: Helena Kloosterman <helena.kloosterman@intel.com>

* Update docs/install_guides/installing-openvino-from-archive-linux.md

Co-authored-by: Helena Kloosterman <helena.kloosterman@intel.com>

* Update docs/install_guides/installing-openvino-from-archive-linux.md

Co-authored-by: Helena Kloosterman <helena.kloosterman@intel.com>

* docs: add link to Release Notes

* docs: Change back to numbered instructions and remove Debian from OS list

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* docs: change wget commands to curl

* docs: change wget commands to curl

* Update docs/install_guides/installing-openvino-from-archive-linux.md

Co-authored-by: Yuan Xu <yuan1.xu@intel.com>

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update installing-openvino-from-archive-linux.md

* Update docs/install_guides/installing-openvino-from-archive-linux.md

Co-authored-by: Helena Kloosterman <helena.kloosterman@intel.com>
Co-authored-by: Yuan Xu <yuan1.xu@intel.com>

EdjeElectronics committed:
Docs: Update "Install OpenVINO Runtime on macOS from Archive File" page (#13347)

* docs: Update intro and step 1

* docs: finish updates

* docs: fix duplicate section

* docs: fix curl command

* docs: clarify archive install is for C++ users

* docs: add link to PyPI install page

* docs: minor fixes

* docs: add link to Release Notes

* docs: Change back to numbered instructions

* docs: typo fix

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Apply suggestions from code review

* Update installing-openvino-from-archive-macos.md

* Update installing-openvino-from-archive-macos.md

* Update installing-openvino-from-archive-macos.md

* Update installing-openvino-from-archive-macos.md

* Update installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

* Update docs/install_guides/installing-openvino-from-archive-macos.md

Co-authored-by: Yuan Xu <yuan1.xu@intel.com>
Co-authored-by: Maciej Smyk <maciejx.smyk@intel.com>

akuporos committed:
[PyOV] Fix building of wheel on windows from cmake (#13795)

ilya-lavrenov committed:
Pdpd tests moved to folder with frontend (#13414)

* Moved pdpd tests to folder with pdpd frontend

* Restored old requirements.txt for paddle temporarily

* FIxed requirements path for pdpd

* Restore old dependency

rkazants committed:
[TF FE] Refactor Pad, PadV2, MirrorPad and add layer tests (#13597)

* [TF FE] Refactor Pad, PadV2, MirrorPad and add layer tests

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

* Recover translators for DynamicStitch

* Apply code-review feedback and workaround for i32 paddings

* Return required type for convert_pad_mode function

* Work around IR reader limitation with i32 type for Pad

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

tgubanova-lohika committed:
[GPU] Add blocked layout support to ExperimentalDetectronTopKRois operation (#12456)

* Addapt ExperimentalDetectronTopKRois to support blocked layouts

* Reformat arg_max_gpu_test

* Add support and tests to arg_max_min 2D case to support for ExperimentalDetectronTopKROIs operation

ilya-lavrenov committed:
Preparation for python debian package (#13800)

Commit:
[GPU] Enable more compiler checks (#13740)

* [GPU] Enable more compiler checks

* [GPU] Disable -Werror for thirdparty

* [GPU] Improved error message when device not found

xuchen-intel committed:
[CPU] Fix issue about gather int32 in Reduce node (#13756)

ilya-lavrenov committed:
Updated zlib to be 1.2.13 (#13803)

EgorDuplensky committed:
[CPU][CORE] Move to the most recent xbyak version (#13753)

Current xbyak has "unresolved external symbols" issue
when built on Windows in 'Debug' (-O0) build type
The issues is fixed in the recent xbyak versions

lohika-denis-kotov committed:
Added -DDNNL_ENABLE_EXCEPTIONS for Debug build (#13672)

ilya-lavrenov committed:
Fully define product version in cmake (#13802)

Lyamin-Roman committed:
[GPU] Pooling refactoring (#13664)

ilya-lavrenov committed:
Removed conda option from python build (#13819)

Commit:
[GPU] Silence ocl include pragma message (#13789)

Lyamin-Roman committed:
[GPU] Fix performance drop on new API (#13767)

p-wysocki committed:
[PyOV] Upgrade ONNX to 1.12.0 (#12847)

* Refactor tests for onnx1.12

* Fix ONNX compatibility and numpy warnings

* Upgrade ONNX to 1.12 in cpp

* Skip unsupported ops

* Revert "Fix ONNX compatibility and numpy warnings"

This reverts commit cd8a0e6b8e68b1f5653a7204f6c6940ce1c38a47.

* Small changes

* Remove unnecessary changes

* Bump Python version in github workflows

* Fix flake error

* Another flake error

* Even more flake errors

* Xfail more tests

* Fix mypy errors

* Change onnx version in tests

* Bump ONNX in MO

* Fix onnx utils in comp

* Minor changes

* Import pathlib

* Add xfails to compatibility tests

* Skip segfaulting tests

* Fix onnx nametuple

* Fix flake

* Try to fix py3.6 onnx

* Fix model importer and reqs

* Add __test__

* Bump OMZ commit

* checkout omz

* Revert omz

* Bump omz

* Bump omz

* Revert omz

* Bump omz

* Change pychecks python version

* Add LayerNorm to expanded ONNX functions

* Apply clang

* Move LayerNorm tests to skip_segfault

* Refactoring

* Minor changes

* Minor changes

* Relax MO reqs

Commit:
Bump pybind11 to v2.10.1 release tag (#13793)

Co-authored-by: Ilya Lavrenov <ilya.lavrenov@intel.com>

songbell committed:
Bell/conditionally enable threadingtests for getavailabledevices (#13772)

* passthrough cache_dir for auto/multi

Signed-off-by: fishbell <bell.song@intel.com>

* enable tests for certain glibc version

Signed-off-by: fishbell <bell.song@intel.com>

* Revert "passthrough cache_dir for auto/multi"

This reverts commit 58591a1bce551a180387b824153302cd2c94ca92.

* clean up

Signed-off-by: fishbell <bell.song@intel.com>

* apply the review comment

Signed-off-by: fishbell <bell.song@intel.com>

Signed-off-by: fishbell <bell.song@intel.com>

sunxiaoxia2022 committed:
add custom setting nstreams nthreads

mryzhov committed:
[GNA] Added support of the all log levels (#13359)

* Added GNA logger

* Added GNA logger class

* Replaced old looger by the new one

* Added support of the all log levels in the tests

* Renaming

* fixed cpplint issue

* Reduced error level

* fixed cpplint issues

* Fixed linking error

* Fixed linux build

* Rebase fix

* Removed unused code

* Fixed compile issue after rebase

* Aligned code usage and style

* Fixed default constructor

* Fixed test

* One more rebase

* Added exveption test

* Fixed the old logging tests

* restored static constants

* Removed dublicated log messages

* Removed incorrect log::level comparation

* Fixed generic api test

* Rebased

rkazants committed:
[TF FE][Tests] Add layer tests for SegmentSum and DynamicPartition in the pre-commit (#13812)

After fixes for TopK and arithmetic operations for integer tensors, we are ready to switch
these tests in the pre-commit

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

ilya-lavrenov committed:
Added per component copyright files for debian / rpm (#13833)

EdjeElectronics committed:
Docs: Update "Install OpenVINO Runtime on Windows from Archive File" page (#13332)

* docs: big update to Windows archive install steps

* docs: apply correct note format

* docs: add link to archives

* docs: minor update

* docs: change archive download link to GitHub

* Update docs/install_guides/installing-openvino-from-archive-windows.md

Co-authored-by: Helena Kloosterman <helena.kloosterman@intel.com>

* docs: typo fix

* docs: minor change

* docs: remove "For Python developers" in Software tab

* docs: fix curl command

* docs: clarify that archive install is for C++ users

* docs: add link to PyPI page

* docs: Change back to numbered instructions

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Apply suggestions from code review

* Update installing-openvino-from-archive-windows.md

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Update docs/install_guides/installing-openvino-from-archive-windows.md

* Update installing-openvino-from-archive-windows.md

Co-authored-by: Helena Kloosterman <helena.kloosterman@intel.com>
Co-authored-by: Yuan Xu <yuan1.xu@intel.com>

yeonbok committed:
Fix error in calling is_dynamic (#13769)

avoskoboinyk-lohika committed:
[CPU] Optimize DFT operation (#11946)

rkazants committed:
[TF FE] Fix output names list for meta format (#13828)

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

usstq committed:
[CPU] Enable scratchpad user mode for oneDNN based nodes (#12038)

l-bat committed:
[POT] Propagate FQ through Gather  (#13583)

* Add Gather to QUANTIZE_AGNOSTIC_OPERATIONS

* Add test

* Update HW configs path in tests

* Fix tests

* Update CI references

Commit:
[GPU] Ability to compile OV with OCL 1.2 (#13706)

tadamowicz committed:
[GNA] Support GNA 3.5 (MTL) target in OV (#13478)

* Enable TLV

* [GNA] Add smoke test for Conv with padding on MTL

* [GNA] Allow skipping tests based on gna lib version

Co-authored-by: Krzysztof Bruniecki <krzysztof.bruniecki@intel.com> 🥇

Commit:
ONNX: Throw exception if attribute reduction is used in ScatterND or ScatterElements (#13778)

* ONNX: Throw exception if `reduction` attribute has unsupported value

Operators:
- ScatterND
- ScatterElements

Wovchena committed:
benchmark_app: --latency_percentile: don't list all values (#13799)

Ticket 95243

Wovchena committed:
Apply improvements I found during https://github.com/openvinotoolkit/openvino/pull/13388 (#13743)

praasz committed:
Review opset1 squeeze for shape inference aspects (#13716)

* Review squeeze operator for
- label preserve and propagation
- partial values preserve and propagation
- interval shapes propagation

* Review static shape inference tests

* Add template shape inference for squeeze

* Update include for OV_EXPECT_THROW

* Correct has_static_axes initialization

* remove test_squeeze from xfail 44968

* Fix inference parameter op with no data
- add additional test to check if axes are parameter without data
- revert xfail marks for squeeze tests

* Fix inference parameter op with no data
- add additional test to check if axes are parameter without data
- revert xfail marks for squeeze tests

* Refactor axes acquire logic

sungeunk committed:
[GPU] add errata for onednn layout selection (#13758)

a-sidorova committed:
[Snippets] Fixed SnippetsMarkSkipped for dynamic branches (#13751)

ilya-lavrenov committed:
Updated exclude patterns for nlohmann json (#13858)

mitruska committed:
GRU/AUGRUCell shape inference function (#13708)

* Add shape_infer function for GRUCell op

* Add shape_infer function for AUGRUCell

* Consts refactor

* Add batch_size check

* Enable GRUCell shape_infer for CPU

* Style apply

* Use OV_EXPECT_THROW in tests

* Use helper for input shapes

* Use .back() instead of index

* Change rnn_seq namespace to rnn

* Fix win warnings

* Enable AUGRUCell/Sequence shape_infer on CPU

* Fix warn

* Fix warn

l-bat committed:
[POT] Unify MatMul scales (#13717)

* Support FQ scales unification for MatMul

* Add test

* Update test_scales.py

Commit:
[PyOV] Fix wheel requirements for new package versions (#13871)

* Bump pybind11 to v2.10.1 release tag

* [PyOV] Fix wheel requirements for new package versions

l-bat committed:
[POT] Update node names in ONNX models (#13872)

p-wysocki committed:
[PyOV] Disallow Python 3.6 usage (#13240)

* Remove py3.6 constraints

* Cleanup out of date python versions

* Bump pybind11 to v2.10.1 release tag

* [PyOV] Fix wheel requirements for new package versions

Co-authored-by: jiwaszki <jan.iwaszkiewicz@intel.com>

wangleis commented on src/inference/src/threading/ie_istreams_executor.cpp:
```diff
@@ -187,9 +187,9 @@ void IStreamsExecutor::Config::SetConfig(const std::string& key, const std::stri
-            //     _streams = LimitHybridStreams(streams.num, _threads);
-            // }
+            if (_threadBindingType == ThreadBindingType::HYBRID_AWARE) {
+                _streams = LimitHybridStreams(streams.num, _threads);
```
Please also do LimitHybridStreams() when use set nthreads after nstreams

wangleis commented on src/inference/src/threading/ie_istreams_executor.cpp:
```diff
@@ -333,15 +333,10 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
-    const auto num_small_cores =
-        custom::info::default_concurrency(custom::task_arena::constraints{}.set_core_type(core_types.front()));
-    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+    const int base_streams_total = parallel_get_max_threads();
```
It may be easy to understand if change `base_streams_total` to `total_threads`

wangleis commented on src/inference/src/threading/ie_istreams_executor.cpp:
```diff
@@ -354,50 +349,55 @@ int IStreamsExecutor::Config::LimitHybridStreams(const int num_streams, const in
                                                          const int num_small_cores) {
-    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
-    const auto num_cores_phys = num_big_cores_phys + num_small_cores;
+    const auto base_streams_total = num_big_cores_phys * 2 + num_small_cores;
```
If `parallel_get_max_threads()` work well on Pcore/Ecore. use `parallel_get_max_threads()` as previous `LimitHybridStreams()`

wangleis commented on src/inference/src/threading/ie_istreams_executor.cpp:
```diff
@@ -354,50 +349,55 @@ int IStreamsExecutor::Config::LimitHybridStreams(const int num_streams, const in
+        // Ensure the balanced use of physical core and logical core, _threads_per_stream_big must be divisible and
+        // smaller than num_big_cores_phys.
+        int threads_per_stream_big = std::min(num_big_cores_phys, std::max(1, base_streams_total / config._streams));
+        config._threads_per_stream_big = (threads_per_stream_big > 1 && threads_per_stream_big % 2 != 0)
```
`threads_per_stream_big` can be odd. 

For example on 6P+4E platform, if customer nstreams is 5, `threads_per_stream_big` and `threads_per_stream_small` are 3. 5 streams include 4 Pcore streams and 1 Ecore streams.

sunxiaoxia2022 committed:
modify comments

rkazants committed:
[TF FE][Common] Use opset Unique in TF FE and move Unique decomposition to common (#13818)

* [TF FE][Common] Move TF FE to opset Unique and add Unique decomposing to common

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

* Add Unique decomposition transformation into the common part

* Add transformation tests for UniqueDecomposition

* Fix unique_decomposition_test

* Simplify condition of transformation applicability

* Clarify the comment when the transformation is applicable

* Fix transformation test

* Fix transformation test

* Add support of opset10 in IR Reader

* Mark the single element case as xfail

* Fix tensor names in the transformation

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

sunxiaoxia2022 committed:
fix code style issue

songbell committed:
Bell/execution device (#13737)

* debug only

Signed-off-by: fishbell <bell.song@intel.com>

* add execution_device property for auto

Signed-off-by: fishbell <bell.song@intel.com>

* Revert "debug only"

This reverts commit 5ef6140b63f121f857564a8c6874201dc44d8c3d.

* revert debug code

Signed-off-by: fishbell <bell.song@intel.com>

* polish tests to cover multiple gpu platform

Signed-off-by: fishbell <bell.song@intel.com>

clean up

Signed-off-by: fishbell <bell.song@intel.com>

* clang format

Signed-off-by: fishbell <bell.song@intel.com>

* only cpu in cpuFunctest

Signed-off-by: fishbell <bell.song@intel.com>

* thread-safe implementation

Signed-off-by: fishbell <bell.song@intel.com>

* remove not needed locl

Signed-off-by: fishbell <bell.song@intel.com>

* apply review comments

Signed-off-by: fishbell <bell.song@intel.com>

* remove blank line

Signed-off-by: fishbell <bell.song@intel.com>

Signed-off-by: fishbell <bell.song@intel.com>

yangwang201911 committed:
Enable AUTO to accept the ov::device::properties with device name contained the index. (#13665)

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

sunxiaoxia2022 committed:
fix threads_per_stream_big is wrong in the some ADL machine which has 6 pcore

yangwang201911 committed:
Fix the logic issue of handling the ov::device::properties in the ie core. (#13482)

* Update the logic to handle the ov:device::properties in the ie core.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* Update test cases.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* 1. Remove the config filter within AUTO plugin when passing the configs to target hardwre device through AUTO plugin.
2. add test cases to check if the secondary properties have been passed into the target device through virtaul device.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* Add mock tests to check if device properties works throgh the AUTO or MULTI  plugin.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* Update.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* Add mock head file for loadnetwork with device properties testing.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* Update.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

* Update gtest commit to fix the warning issue appeared in the macro MATCHER_P.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

Commit:
[GPU] Remove f16<->f32 conversion duplicates (#13735)

p-wysocki committed:
Bump pytorch (#13688)

mvafin committed:
Fix Slice issue in MO IR Reader (#13784)

* Fix slice issue in MO IR Reader

* Add unit test

* Fix slice test

sunxiaoxia2022 committed:
add one condition that custom set nthreads but not set nstreams

byungilm committed:
[GPU] Resolve reduce test failure (#13607)

+ Added logic to select planar format for b-f axes reduction
+ Bugfix of reorder_unreduced_axis_no_fusion
+ Bugfix to resolve an issue of cgan_mnist

Signed-off-by: Min, Byungil <byungil.min@intel.com>

rkazants committed:
[TF FE] Refactor SpaceToDepth and test it in the precommit (#13882)

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

rkazants committed:
[TF FE] Refactor Where translator and test it (#13880)

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

akuporos committed:
[Pyskip failed tests (#13892)

Lyamin-Roman committed:
[GPU] AvgPool/MaxPool dynamic support (#13754)

l-bat committed:
[POT] Skip unstable test (#13889)

praasz committed:
Review opset1 split for shape inference aspects (#13844)

* Review split operator for shape inference:
- check label and dimension propagation
- check evaluation lowe, upper and labels

* Review split static shape inference by tests

* Use shape_infer in split evaluate

* Fix compilation issues

* Fix usage of broadcast in bound tests

venki-thiyag committed:
#13888 [Bug] Unable to load OpenVINO in Electron app node addon (#13890)

Co-authored-by: venki <venkatesan.t@ringcentral.com>

yeonbok committed:
[GPU] Enable Conv/GroupConv shape infer for explicit/valid padding (#13731)

* Enabled Conv dynamic shape & Added conv dynamic functional test

* Fix layout optimizor

* Revert layout optimizer change

* Set expected layout of convolution for dynamic shape to be fsv16

* Fixed groupConv for dynamic shape

Added 1D dw/group convolution tests for both static & dynamic
Currently, 1D group convolution is only works for dw..

* Minor fix

* Updated for 3D conv dynamic case + corresponding test

* Fix lint error

* Fix INSTANTIATE_TEST_CASE to INSTANTIATE_TEST_SUITE

* Fix clang format & skip 1D group conv test

ilya-lavrenov committed:
temporary commented failed test (#13904)

andrew-k-park committed:
[GPU] Fix performance degradation of bfyx conv with FP16 compared to FP32 (#13847)

Signed-off-by: Andrew Park <andrew.park@intel.com>

Signed-off-by: Andrew Park <andrew.park@intel.com>

tsavina committed:
DOCS: Developer documentation Inference component update (#13895)

* dev docs fixes

* fix space

* fix paragraphs

ilyachur committed:
Add ctest for some test targets (#13727)

* Add ctest for some test targets

* Fixed test coverage

* Replaced unit tests cmake script

* Fixed code style

* Use ov_add_test_target for FEs

* Fixed code style

ilyachur committed:
Enable disabled check (#13908)

sirzabek committed:
GNA MTL mixed precision (#13166)

* [GNA] Enable int8 weights for convolutions

* [GNA] Implement convolution low precision tests for MTL

Co-authored-by: Marcin Kacprzak <marcin.kacprzak@intel.com>

ilya-lavrenov committed:
Risc-v toolchain (#13897)

p-wysocki committed:
[PyOV] Enable Python 3.10 on Azure CI (#12578)

* Fix mypy for numpy==1.21.6

* Apply CR changes

* Allow Python 3.6 for open source

* Attempt to silence mypy

* Revert to 3.6 in setup.pys

* Fix merge conflict

* Add newline

* Fix Python version assert

* Test Python 3.10 CI

* Try bumping up pybind to 2.10

* Link default python to python 3.10

* Add sudo to changing default python

* Bump python version in setupvars

* Fix exception in test

* Do the same for compatibility

* Apply same CI changes to other pipelines

* Attempt to fix arm and windows CIs

* Fix win py lib version and links?

* Try to fix CI

* Fix win python path

* Update win path

* Test disable PDPD

* Disable TF for testing CI

* Add sudo to rm

* Test linux

* Update requirements

* RM sudo

* Add debug step to CI on ARM

* build py from source on linux

* Remove debug step from ARM

* Fix err

* Fill num_proc

* Fic py installation

* Bump pdpd to 2.3.1

* Bump np vers

* Try to upgrade onnx to 1.12.0

* Revert onnx upgrade, enable frontends

* Build py shared

* Export library

* activate py env

* Bump scipy for py3.10

* Another scipy bump

* Fix Windows path, add nfs helper for mount

* Attempt to fix win_cc

* Batch of changes

* Try to fix ngraph-onnx-lin

* Install lsb-release for docker on ngraph-onnx-lin

* Link lsb_release

* Try bumping Ubuntu to 22.04

* Revert ubuntu change in dockerfile

* Refactor tests for onnx1.12

* Fix ONNX compatibility and numpy warnings

* Refactor tests for onnx1.12

* Fix ONNX compatibility and numpy warnings

* Upgrade ONNX to 1.12 in cpp

* Skip unsupported ops

* Revert "Fix ONNX compatibility and numpy warnings"

This reverts commit cd8a0e6b8e68b1f5653a7204f6c6940ce1c38a47.

* Small changes

* Remove unnecessary changes

* Bump Python version in github workflows

* Fix flake error

* Another flake error

* Even more flake errors

* Xfail more tests

* Fix mypy errors

* Change onnx version in tests

* Bump ONNX in MO

* Fix onnx utils in comp

* Minor changes

* Import pathlib

* Add xfails to compatibility tests

* Skip segfaulting tests

* Fix onnx nametuple

* Fix flake

* Try to fix py3.6 onnx

* Switch to venv in windows

* Fix model importer and reqs

* Add __test__

* Debug print

* Fix flake

* Update linux.yml

* Update windows.yml

* Update linux.yml

* Update windows.yml

* Install python3.10

* Update windows.yml

* Update windows.yml

* Update windows.yml

* Update windows.yml

* Update windows.yml

* Update windows.yml

* Update windows.yml

* Update windows.yml

* Install py embedded

* Fix dir

* Fix curl

* Add complete file

* CP file

* Bump py ver

* Fix missing venv

* Try to install

* Disable venv

* Bump OMZ commit

* checkout omz

* Revert omz

* Bump omz

* Revert pybind bump

* Update .gitmodules

* Change pybind to fork

* change ssize_t to size_t

* Install bz2 for py build

* Bump omz

* Bump pandas requirement version

* Revert omz

* Install missing clang

* Bump omz

* Bump versions on fork

* Bump numpy in model zoo

* Fix clang

* Skip segfaulting pdpd test

* Skip another pdpd test

* Skip pdpd places.cpp

* Bump py ver in setupvars.bat

* Disable paddle tests

* Use int instead of str

* Fix requirements path

* Enable lin-cc

* Fixes in CI

* lin-cc and ngrpah-onnx changes

* Update num_proc

* Update ngraph-onnx

* Minor change

* Fix env var

* Change ubuntu to 20

* Get compiler on ngraph-onnx

* Add sudo to apt-get

* Add SSL

* Download ssl

* Different way of installing ssl

* Minor changes

* install zlib

* Fix env var

* Fix docker building error

* Revert changes

* Modify dockerfile

* Change pychecks python version

* Cleanup

* Add LayerNorm to expanded ONNX functions

* Revert previous commit

* Add LayerNorm to expanded ONNX functions

* Apply clang

* Move LayerNorm tests to skip_segfault

* Unskip pdpd tests

* Enable pdpd tests on linux

* Refactoring

* Remove 3.6 from reqs

* Refactoring

* Minor changes

* Minor changes

* Relax MO reqs

* Bump torch version

* relax pytest

* Specify pytest version

* Fix pytest pip conflict

* Bump pybind11 to v2.10.1 release tag

* Specify separate pytest version for py3.10

* Bump attrs

* Add py dependency

* Refactor versions

* Add venv activation to tf tests

* Fix fdupes

* Bump OMZ submodule

* Remove find and replace

* Change pybind to upstream fork

* Set pybind to proper commit

* Minor changes

* Try to fix scipy

* Updated exclude patterns for nlohmann json

* [PyOV] Fix wheel requirements for new package versions

* Specify CPU plugin versions

* Bump OMZ version

* Fix linter

* Fix linter 2

* Fix linter 3

* Disable linter

Co-authored-by: Mateusz <mateusz.mikolajczyk@intel.com>
Co-authored-by: jiwaszki <jan.iwaszkiewicz@intel.com>
Co-authored-by: Ilya Lavrenov <ilya.lavrenov@intel.com>

sunxiaoxia2022 committed:
remove the limit to nstreams

sunxiaoxia2022 committed:
remove the feature in branch p_e_core_phase2_logic

ilya-lavrenov committed:
Added option to disable compile_tool (#13903)

* Updated pugixml

* Added option to disable compile_tool

sunxiaoxia2022 committed:
add comments

nshchego committed:
[CPU] GridSample operation implementation. (#12910)

Commit:
Fix datetime not found issue in utils (#13690)

Commit:
Introduce PullThroughReduce and ReduceReshapeFusion transformations (#13047)

* Initial version of PullThroughReduce and ReduceReshapeFusion transformation

* remove std namespace

* headers + namespaces clean-up

* resolved problem with ambiguous namespace

* clang format applied

* tests refactor. part 1

* tests refactor. part.2

* changed includes

* review remarks

* tests refactor. part.3

* review remarks

* fix build problem

* remarks

* fix reserving memory

* review remarks

* styles applied

* handle axes in the middle in try_get_unsqueeze_axes_from_reshape

* handle case of scalar input shape

Co-authored-by: Michal Lukaszewski <michal.lukaszewski@intel.com>

Commit:
[PyOV] Fix mypy after bump, general refactor (#13913)

* Fix mypy errors after bump PEP482, reformat code in general

* Enable checks

* Fix smaller issues

* Revert typo with comments

* Fix indents

rkazants committed:
[TF FE] Refactor SpaceToBatchND and BatchToSpaceND operations (#13884)

* [TF FE] Refactor SpaceToBatch and BatchToSpace operations

Make them to support inputs of dynamic rank and shapes

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

* Add layer tests in the pre-commit

* Fix for normalizing padding and crops

* Correct the type of paddings and crops

* Fix incorrect changes for paddings and crops

* Update src/frontends/tensorflow/src/op/space_to_batch_nd.cpp

Co-authored-by: Maxim Vafin <maxim.vafin@intel.com>

* Update src/frontends/tensorflow/src/op/space_to_batch_nd.cpp

Co-authored-by: Maxim Vafin <maxim.vafin@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>
Co-authored-by: Maxim Vafin <maxim.vafin@intel.com>

sunxiaoxia2022 committed:
Xiaoxia/Default setting with hyper thread enabled on Hybrid-aware P/E-core (#13773)

* add HybridAware stream setting and core binding

* fix clang format issue

* unified code style, add parameter check

* correct input affinity skip NUMA, modify function name

* remove unnecessary floor

* fix ci compile issue on Mac/windows platform

* modify smoke_SetConfigAffinity test

* modify ov_core_set_and_get_property_enum test, affinity HYBRID_AWARE is changed to NUMA

* remove affinity correcting on this pr

* revert ov_core_test.cpp

* merge function by comments

* fix code style issue

* add custom nstreams setting, remove mutable qualifier

* fix code style issue

* fix some issues according to comments

* modify UpdateHybridCustomThreads

* fix code style issue

* modify comments

* use logic core in phase 2

* modify steams threads setting method

* move properties used in P/E-core to ie_internal_plugin_config.hpp

* fix streams is wrong on P/E-core which has 6 big cores

* modify code according to comments

* add threads_per_stream_small=3

* add the judgment of logical core

* fix code style issue

Co-authored-by: Shen, Wanglei <wanglei.shen@intel.com>

mitruska committed:
Broadcast-3 shape_infer review (#13849)

* v3::Broadcast tests

* Shape infer improvements

* Use Dimension broadcast_merge in Bidirectional mode

* More tests

* Style apply

* Fix Unqueeze with Broadcast tests

* Align Squeeze/Unsqueeze tests

* Fix warning

* Updates test shapes names

* loops refactor

* Remove redundant set/get labels from expected shapes

* Padding refactor

* Tests refactor

* Revert auto for i in loop

rkazants committed:
[Core] Fix StridedSlice shape infer in case dynamic end and input of dynamic shape (#13909)

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

ahnyoung-paul committed:
[GPU] Fix multistream issue for dynamic shape (#13433) (#13433)

- Separate kernels_cache::add_kernel from factory (choose_impl)
- Reset kenrels_cache._kernels after kernels_cache.compile
- Create cldnn unit test case to check multi-stream processing

tgubanova-lohika committed:
[GPU] Add blocked layouts support to normalize operation (#12459)

* Add blocked layouts support to normalize operation

* Add normalize logic for blocked and plain layouts to one kernel

ceciliapeng2011 committed:
[CPU] add support for internal ops augrucell and augrusequence (#12647)

* [cpu] enable augru.

* initiate opset_private_tbl.hpp for private ops.

* fix ngraph_functions target_linker.

* temporaliy disable a failure test.

byungilm committed:
[GPU] Reduce overhead in device_query (#13200)

+ Reduced overhead of is_local_block_io_supported in ocl_device
+ Updated hash_value in kernels_cache

Signed-off-by: Min, Byungil <byungil.min@intel.com>

sungeunk committed:
fuse reorder to next conv (#13683)

-in_fmt:bfyx, out_fmt:b_fs_yx_fsv16, bs_fs_yx_bsv16_fsv16
-shallow input conv

sunxiaoxia2022 committed:
fix conflicts

sunxiaoxia2022 committed:
rename num_small_cores to num_small_cores_phys

sunxiaoxia2022 committed:
fix code style issue

sunxiaoxia2022 committed:
fix thread number is wrong in corner case on 6Pcore and 10Pcore condition

sunxiaoxia2022 committed:
fix some condition which nstreams<2

wangleis committed:
add check for config._threads_per_stream_big

wangleis committed:
fix code style issue
